src/ur/workday.ur

```diff
@@ -0,0 +1,378 @@
+     Preview : string
+}
+val _ : json relatedFeedback =
+    json_record {Id = "id",
+                 Preview = "descriptor",
+                 BusinessProcessParameters = "businessProcessParameters"}
+
+type anytimeFeedback = {
+     Id : wid,
+     Preview : option string,
+     ToWorker : option descriptor, (* who the feedback is about *)
+     WorkersToNotify : option (list descriptor), (* always the direct manager *)
+     BusinessProcessParameters : option businessProcessParameters,
+     HiddenFromWorker : option bool, (* True if the feedback event is confidential between the feedback provider and the manager. Workers don't see confidential feedback. *)
+     HiddenFromManager : option bool, (* True if the feedback event is private between the feedback provider and the worker. Private feedback isn't shared with managers. *)
+     Comment : option string, (* actual feedback *)
+     FromWorker : option descriptor, (* Identity of the responder. Only provided if the request asked to know the identity of the responder. *)
+     OverallStatus : option string,
+     GivenDate : option time,
+     ShowFeedbackProviderName: option bool,
+     (* unused parameters: *)
+     (*
+     Badge : descriptor,
+     Href : string
+     AlsoAbout : list descriptor, (* other workers the feedback is about *)
+     RelatedFeedbackEvents : list relatedFeedback,
+     RelatesTo : descriptor,
+      *)
+}
+val json_anytimeFeedback : json anytimeFeedback =
+    json_record_withOptional
+        {Id = "id"}
+        {Preview = "descriptor",
+         ToWorker = "toWorker",
+         WorkersToNotify = "workersToNotify",
+         BusinessProcessParameters = "businessProcessParameters",
+         HiddenFromWorker = "hiddenFromWorker",
+         HiddenFromManager = "hiddenFromManager",
+         Comment = "comment",
+         FromWorker = "fromWorker",
+         OverallStatus = "overallStatus",
+         GivenDate = "feedbackGivenDate",
+         ShowFeedbackProviderName = "showFeedbackProviderName"
+                 (*
+                 RelatesTo = "relatesTo",
+                 Badge = "badge",
+                 AlsoAbout = "feedbackAlsoAbout",
+                 RelatedFeedbackEvents = "relatedFeedbackEvents",
+                  *)
+                }
+
+datatype feedbackStatus = Complete | Requested
+(*
+val show_feedbackStatus : show feedbackStatus =
+ fn s => case s of
+             Complete => "Complete"
+           | Requested => "Requested"
+val read_feedbackStatus : read feedbackStatus
+val eq_feedbackStatus : eq feedbackStatus
+*)
+
+con feedback = [
+      Id = wid,
+      Created = time,
+      RequestedDate = option time,
+      About = wid, (* who the feedback is about *)
+      Provider = wid, (* who writes the feedback *)
+      RequestedBy = option wid,
+      Body = string, (* response *)
+      Guidance = string,
+      Status = string, (* TODO: use feedbackStatus *)
+      SendMail = bool,
+      LastUpdated = time
+]
+
+con feedbackWithoutId = [
```

Comment:
Ah, one perhaps-justifiable move would be to use this constructor to build both record types (with and without ID) without duplication, but in similar contexts, I have instead just made ID optional, as far as the Ur/Web type system knows.  Of course you can enforce its presence with value-level code, as needed.

Reply:
Making the ID optional is probably fine and I have gone for that solution in my newest changes (soon to be pushed), but it is worth noting that the ID is used as a primary key for our feedback tables, but new rows do not receive an ID until they are POSTed to Workday. We had a conversation about this a while ago and proposed creating Nectry-specific UUIDs for these objects that are uniquely associated with Workday IDs. We should settle on a best practice soon.

Reply:
We decided to go for a Nectry-specific UUID. Now the Workday IDs are called `Wid` and the Nectry IDs are `Id`.